src/mesh/mesh_refinement.cpp

```diff
@@ -896,67 +932,96 @@ void MeshRefinement::ProlongateSharedFieldX2(
   Coordinates *pco = pmb->pcoord;
   int fsi = (si - pmb->cis)*2 + pmb->is, fei = (ei - pmb->cis)*2 + pmb->is + 1;
   if (pmb->block_size.nx3 > 1) {
-    for (int k=sk; k<=ek; k++) {
-      int fk = (k - pmb->cks)*2 + pmb->ks;
-      const Real& x3m = pcoarsec->x3s2(k-1);
-      const Real& x3c = pcoarsec->x3s2(k);
-      const Real& x3p = pcoarsec->x3s2(k+1);
-      Real dx3m = x3c - x3m;
-      Real dx3p = x3p - x3c;
-      const Real& fx3m = pco->x3s2(fk);
-      const Real& fx3p = pco->x3s2(fk+1);
-      for (int j=sj; j<=ej; j++) {
-        int fj = (j - pmb->cjs)*2 + pmb->js;
-        if (std::strcmp(COORDINATE_SYSTEM, "spherical_polar") == 0) {
+    if (fluxinterp) {
+      for (int k=sk; k<=ek; k++) {
+        int fk = (k - pmb->cks)*2 + pmb->ks;
+        const Real& x3m = pcoarsec->x3s2(k-1);
+        const Real& x3c = pcoarsec->x3s2(k);
+        const Real& x3p = pcoarsec->x3s2(k+1);
+        Real dx3m = x3c - x3m;
+        Real dx3p = x3p - x3c;
+        const Real& fx3m = pco->x3s2(fk);
+        const Real& fx3p = pco->x3s2(fk+1);
+        for (int j=sj; j<=ej; j++) {
+          int fj = (j - pmb->cjs)*2 + pmb->js;
           pco->Face2Area(fk,   fj,  fsi, fei, sarea_x2_[0][0]);
           pco->Face2Area(fk+1, fj,  fsi, fei, sarea_x2_[1][0]);
           pcoarsec->Face2Area(k, j, si, ei, csarea_x2);
-        }
-        for (int i=si; i<=ei; i++) {
-          int fi = (i - pmb->cis)*2 + pmb->is;
-          const Real& x1m = pcoarsec->x1s2(i-1);
-          const Real& x1c = pcoarsec->x1s2(i);
-          const Real& x1p = pcoarsec->x1s2(i+1);
-          Real dx1m = x1c - x1m;
-          Real dx1p = x1p - x1c;
-          const Real& fx1m = pco->x1s2(fi);
-          const Real& fx1p = pco->x1s2(fi+1);
-          Real ccval = coarse(k,j,i);
-
-          Real gx1m = (ccval - coarse(k,j,i-1))/dx1m;
-          Real gx1p = (coarse(k,j,i+1) - ccval)/dx1p;
-          Real gx1c = 0.5*(SIGN(gx1m) + SIGN(gx1p))*std::min(std::abs(gx1m),
-                                                             std::abs(gx1p));
-          Real gx3m = (ccval - coarse(k-1,j,i))/dx3m;
-          Real gx3p = (coarse(k+1,j,i) - ccval)/dx3p;
-          Real gx3c = 0.5*(SIGN(gx3m) + SIGN(gx3p))*std::min(std::abs(gx3m),
-                                                             std::abs(gx3p));
-          if (std::strcmp(COORDINATE_SYSTEM, "spherical_polar") == 0) {
+          for (int i=si; i<=ei; i++) {
+            int fi = (i - pmb->cis)*2 + pmb->is;
+            const Real& x1m = pcoarsec->x1s2(i-1);
+            const Real& x1c = pcoarsec->x1s2(i);
+            const Real& x1p = pcoarsec->x1s2(i+1);
+            Real dx1m = x1c - x1m;
+            Real dx1p = x1p - x1c;
+            const Real& fx1m = pco->x1s2(fi);
+            const Real& fx1p = pco->x1s2(fi+1);
+            Real ccval = coarse(k,j,i);
             Real dfx1 = fx1p - fx1m;
             Real dfx3 = fx3p - fx3m;
```

Comment:
dfx3 does not depends on i and j, so this line can be moved out of the i,j loops.